lib/core/filtering/simple_rule.rb

```diff
@@ -34,26 +36,50 @@ class Rule
         GREATER_THAN = '>'
       end
 
-      attr_reader :policy, :field, :rule, :value, :id
+      attr_reader :policy, :field, :rule, :value, :id, :order
 
       def initialize(rule_hash)
-        @policy = rule_hash.fetch(POLICY)
-        @field = rule_hash.fetch(FIELD)
-        @rule = rule_hash.fetch(RULE)
-        @value = rule_hash.fetch(VALUE)
-        @id = rule_hash.fetch(ID)
+        @policy = SimpleRule.flex_fetch(rule_hash, POLICY)
+        unless @policy == Policy::INCLUDE || @policy == Policy::EXCLUDE
+          raise "Invalid policy '#{policy}' for rule '#{rule_hash}'"
+        end
+        @field = SimpleRule.flex_fetch(rule_hash, FIELD)
+        @rule = SimpleRule.flex_fetch(rule_hash, RULE)
+        @value = SimpleRule.flex_fetch(rule_hash, VALUE)
+        @id = SimpleRule.flex_fetch(rule_hash, ID)
+        @order = SimpleRule.flex_fetch(rule_hash, ORDER, 0)
```

Comment:
If we're going to add `order` here, I suggest we require it, and not provide a default value. Otherwise we risk non-deterministic ordering in specs if we don't have an explicit order field defined and they all get `:order => 0` from this default.

Reply:
My thinking was that if we don't have order, then the ordering probably doesn't matter (like it doesn't now for pre-processing) and we can safely assume some default value. But you're probably right.